test/python/apps/exchange.py

```diff
@@ -76,6 +77,30 @@ class Payload_Encoding(Enum):
 )
 
 
+@dataclass
+class Identity:
+    privkey: ec.EllipticCurvePrivateKey
+    credentials: bytes
+    signed_credentials: bytes
+
+def generate_partner_keys(curve, partner_name: bytes = b"Default_Partner") -> Identity:
```

Comment:
In term of namespace, this function could either be considered as a constructor for `Identity`, and so could be an `__init__` or a factory `@staticmethod` of `Identity`, or a helper for `ExchangeClient` (and be a method of it).

I believe it would make sense for this to be used outside of `Exchange`, so I'm in favor of embedding it in `Identity` (and `Identity` can be moved to a common place - like `utils.py`? Not a big fan of such generic module, but maybe we need a bit more time to decide how to manage this kind of struct/helpers)

Reply:
I've redone the whole "Identity submodule" into something more clear
I'm not sure it should be made generic, it does not really have a meaning outside of an Exchange transaction
I'll let you see the proposal and decide accordingly